src/components/DayPicker.jsx

```diff
@@ -753,6 +751,46 @@ class DayPicker extends React.Component {
 
     const isHorizontalAndAnimating = this.isHorizontal() && isCalendarMonthGridAnimating;
 
+    const calendarInfoPositionTop = calendarInfoPosition === INFO_POSITION_TOP;
+    const calendarInfoPositionBottom = calendarInfoPosition === INFO_POSITION_BOTTOM;
+    const calendarInfoPositionLeft = calendarInfoPosition === INFO_POSITION_LEFT;
+    const calendarInfoPositionRight = calendarInfoPosition === INFO_POSITION_RIGHT;
+    const calendarInfoIsInline = calendarInfoPositionLeft || calendarInfoPositionRight;
+
+    const calendarInfo = renderCalendarInfo && (
+      <div
+        ref={this.setCalendarInfoRef}
+        {...css((calendarInfoIsInline) && styles.DayPicker_calendarInfo__horizontal)}
+      >
+        { renderCalendarInfo() }
+      </div>
+    );
+
+    const calendarInfoWith = renderCalendarInfo && calendarInfoIsInline
+      ? calculateDimension(this.calendarInfo, 'width', true, true)
+      : 0;
+
+    const firstVisibleMonthIndex = this.getFirstVisibleIndex();
+    const wrapperHorizontalWidth = (calendarMonthWidth * numberOfMonths) + (2 * DAY_PICKER_PADDING);
+    const fullHorizontalWidth = wrapperHorizontalWidth + calendarInfoWith;
+
+    const transitionContainerStyle = {
+      width: this.isHorizontal() && wrapperHorizontalWidth,
+      height,
+    };
+
+    const dayPickerWrapperStyle = {
+      width: wrapperHorizontalWidth,
+    };
+
+    const dayPickerStyle = {
+      width: this.isHorizontal() && fullHorizontalWidth,
+
+      // These values are to center the datepicker (approximately) on the page
+      marginLeft: this.isHorizontal() && withPortal && -fullHorizontalWidth / 2,
+      marginTop: this.isHorizontal() && withPortal && -calendarMonthWidth / 2,
```

Comment:
rather than overloading `&&` for value selection, can these use explicit ternaries?

Reply:
I changed it, but for the record, the implicit ternaries were already here before, I only change their position in the code..

Reply:
Fair enough :-) thanks for leaving it cleaner than you found it!

Reply:
My pleasure ! :)